app/src/main/java/com/isc/hermes/controller/IncidentFormController.java

```diff
@@ -57,6 +70,25 @@ private void setButtonsOnClick(){
             incidentForm.setVisibility(View.GONE);
             mapController.deleteMarks();
             Toast.makeText(context, "Incident Saved Correctly.", Toast.LENGTH_SHORT).show();
+
+            @SuppressLint("StaticFieldLeak") AsyncTask<Void, Void, Integer> task = new AsyncTask<Void, Void, Integer>() {
+                @Override
+                protected Integer doInBackground(Void... voids) {
+                    return uploadIncidentDataBase();
+                }
+
+                @Override
+                protected void onPostExecute(Integer responseCode) {
+                    if (responseCode == HttpURLConnection.HTTP_OK) {
+                        Toast.makeText(context, "Incident uploaded successfully.", Toast.LENGTH_SHORT).show();
+                    } else {
+                        Toast.makeText(context, "Incident could not be uploaded.", Toast.LENGTH_SHORT).show();
+                    }
+                }
+            };
+
+            // Execute the task
+            task.execute();
         });
     }
```

Comment:
Method setButtonsOnClick is too large, you should refactor it..

Reply:
This observation has been resolved.